### PR TITLE
rsp[:] = const_value, used in constant initializer for rsp weight

### PR DESCRIPTION
**Pull request by ZiyueHuang**
## Description ##
can be used in constant initializer for row_sparse weight

cc @eric-haibin-lin 

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] `rsp[:] = const_value`, unittest

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/python/unittest/test_init.py [hunk 1]
```diff
@@ -44,8 +44,21 @@ def test_aux_init():
     assert (mod.get_params()[1]['bn_moving_var'].asnumpy() == 1).all()
     assert (mod.get_params()[1]['bn_moving_mean'].asnumpy() == 0).all()
 
+def test_rsp_const_init():
+    shape = (10, 100)
```
Comment (eric-haibin-lin) [inline tests/python/unittest/test_init.py]: nit: shape=`(10,10)` should be sufficient for unit test to verify correctness

Comment (eric-haibin-lin) [reply tests/python/unittest/test_init.py]: Can we also add test `initializer.Zero` and `initializer.One` to make it more complete?

File: src/ndarray/ndarray.cc [hunk 2]
```diff
@@ -309,29 +309,56 @@ void SetValueOp(const real_t &rhs, NDArray *out) {
...
-        }, ret.ctx(), {}, {ret.var()},
-        FnProperty::kNormal, 0, PROFILER_MESSAGE_FUNCNAME);
-      break;
-    }
+  const NDArrayStorageType stype = ret.storage_type();
+  if (stype == kDefaultStorage) {
+    switch (ret.ctx().dev_mask()) {
+      case cpu::kDevMask: {
+        Engine::Get()->PushSync([rhs, ret](RunContext ctx) {
+            CHECK(ret.storage_type() == kDefaultStorage);
```
[the remark this answers is not in the document]
Comment (ZiyueHuang) [reply src/ndarray/ndarray.cc]: Done.

File: src/ndarray/ndarray_function.cc [hunk 3]
```diff
@@ -183,5 +183,13 @@ void ElementwiseSum<cpu>(mshadow::Stream<cpu>* s,
   }
 }
 
+
+template<>
+void SetValueRsp<cpu>(mshadow::Stream<cpu> *s,
```
Comment (eric-haibin-lin) [inline src/ndarray/ndarray_function.cc]: Would it be better for `ndarray.cc` to call `SetValue(stream, real_t, NDArray)`, then `SetValue` internally checks the stype and calls `SetValueRsp`?

**Comment by ZiyueHuang** [comment]
@piiswrong

**Pull request merged**